frontend/packages/kubevirt-plugin/src/k8s/wrapper/common/k8s-resource-wrapper.ts

```diff
@@ -1,10 +1,48 @@
 /* eslint-disable lines-between-class-members */
 import { getName, hasLabel, getLabels } from '@console/shared/src';
-import { K8sResourceKind } from '@console/internal/module/k8s';
+import { apiVersionForModel, K8sKind, K8sResourceKind } from '@console/internal/module/k8s';
 import { Wrapper } from './wrapper';
+import { K8sResourceKindMethods } from '../types/types';
 
-export class K8sResourceWrapper<RESOURCE extends K8sResourceKind> extends Wrapper<RESOURCE> {
+export class K8sResourceWrapper<
+  RESOURCE extends K8sResourceKind,
+  SELF extends K8sResourceWrapper<RESOURCE, SELF>
+> extends Wrapper<RESOURCE, SELF> implements K8sResourceKindMethods {
   getName = () => getName(this.data);
   getLabels = (defaultValue = {}) => getLabels(this.data, defaultValue);
   hasLabel = (label: string) => hasLabel(this.data, label);
+
+  setModel = (model: K8sKind) => {
+    this.data.kind = model.kind;
+    this.data.apiVersion = apiVersionForModel(model);
+    return (this as any) as SELF;
+  };
+
+  setName = (name: string) => {
```

Comment:
`K8sResourceObjectWithTypePropertyWrapper` and `K8sResourceWrapper` cannot be united so we have to duplicate code here. But it should be in only these two places and the duplicated logic is pretty simple

Reply:
Class hierarchies impose limits on reusability from inheritance perspective.

The closest common ancestor of `K8sResourceWrapper` and `K8sResourceObjectWithTypePropertyWrapper` is `Wrapper`. But IIUC it doesn't make sense to put these methods into `Wrapper`.

One way to reuse duplicated code here would be the concept of mixins. (Not a big deal if the logic is simple enough.)

Reply:
yes, let's keep this in mind and take care of it if the logic becomes too advanced or diverges in the future